scripts/Test-RepoPackage.ps1

```diff
@@ -0,0 +1,633 @@
+    [int]$screenshotTimeout,
+    [Parameter(Mandatory = $true)]
+    [bool]$takeScreenshot,
+    [Parameter(Mandatory = $true)]
+    [string]$screenShotDir
+  )
+
+  $dependentSource = GetPackagePath -packageName $package.DependentPackage
+  if ($dependentSource) {
+    try {
+      RunChocoPackProcess -path $dependentSource | WriteOutput
+    } catch {
+      WriteOutput $_ -type Error
+      return $package.Name
+    }
+    $packageNames = @($package.Name ; $package.DependentPackage)
+  } else {
+    $packageNames = @($package.Name)
+  }
+
+  $arguments = @{
+    timeout = $chocoCommandTimeout
+    takeScreenshot = $takeScreenshot
+    timeoutBeforeScreenshot = $screenshotTimeout
+    arguments = @('uninstall';$packageNames)
+    screenShotDir = $screenShotDir
+  }
+  try {
+    if (!(RunChocoProcess @arguments)) { return $package.Name }
+  } catch {
+    WriteOutput $_ -type Error
+    return $package.Name
+  }
+
+  return ""
+}
+
+function TestUninstallAllPackages() {
+<#
+.SYNOPSIS
+  Function responsible for running the uninstall tests on the specified packages.
+  But only if the package names isn't listed in the specified failedInstall object array.
+#>
+  param(
+    [Parameter(Mandatory = $true)]
+    $packages,
+    [Parameter(Mandatory = $true)]
+    [int]$chocoCommandTimeout,
+    [Parameter(Mandatory = $true)]
+    [int]$screenshotTimeout,
+    [Parameter(Mandatory = $true)]
+    [bool]$takeScreenshot,
+    [Parameter(Mandatory = $true)]
+    [string]$screenShotDir,
+    [object[]]$failedInstalls,
+    [Parameter(ValueFromRemainingArguments = $true)]
+    [object[]]$ignoredValues
+  )
+
+  $packages | % {
+    $name = $_.Name
+    $packageFailed = $failedInstalls | ? { $_ -eq $name }
+    if ($packageFailed) {
+      WriteOutput "$name failed to install, skipping uninstall test..." -type Warning
+      return ""
+    } else {
+      pushd $_.Directory
+      UninstallPackage `
+        -package $_ `
+        -chocoCommandTimeout $chocoCommandTimeout `
+        -screenshotTimeout $screenshotTimeout `
+        -takeScreenshot $takeScreenshot `
+        -screenShotDir $screenShotDir
+      MoveLogFile -packageName $name -commandType 'uninstall'
+      popd
+    }
+  } | ? { $_ -ne $null -and $_ -ne '' }
+}
+
+if ($packageName) {
+  $packages = GetPackagesFromName -packageName $packageName
+} else {
+  $packages = GetPackagesFromDiff -diffAgainst 'origin/master'
+}
+
+$packages = RemoveDependentPackages -packages $packages
+
+if ($CleanFiles) {
+    CleanFiles -screenShotDir $screenShotDir
+}
+
+if (!$packages) {
+  WriteOutput "No changed packages was found. Exiting tests..." -type Warning
+  return;
+}
+
+if ($TakeScreenshots) { . "$PSScriptRoot\Take-ScreenShot.ps1" }
+$arguments = @{
+  packages = $packages
+  chocoCommandTimeout = $chocoCommandTimeout
+  takeScreenshot = $TakeScreenshots
+  screenShotTimeout = $timeoutBeforeScreenshot
+  screenShotDir = $screenShotDir
+}
+
+switch -Exact ($type) {
```

Comment:
This better not be switch, the code is not DRY because of it.
Better:

```
if ('update','all' -contains $type) { ... }
if ('install','all' -contains $type) { ... }
if ('unintsall', 'all' -contains $type) { }
```